exporters/otlp/src/otlp_http_exporter.cc

```diff
@@ -53,7 +74,38 @@ opentelemetry::sdk::common::ExportResult OtlpHttpExporter::Export(
 
   proto::collector::trace::v1::ExportTraceServiceRequest service_request;
   OtlpRecordableUtils::PopulateRequest(spans, &service_request);
-  return http_client_->Export(service_request);
+  std::size_t span_count = spans.size();
+#ifdef ENABLE_ASYNC_EXPORT
+  http_client_->Export(
+      service_request, [span_count](opentelemetry::sdk::common::ExportResult result) {
+        if (result != opentelemetry::sdk::common::ExportResult::kSuccess)
+        {
+          OTEL_INTERNAL_LOG_ERROR("[OTLP HTTP Client] ERROR: Export "
+                                  << span_count
+                                  << " trace span(s) error: " << static_cast<int>(result));
+        }
+        else
+        {
+          OTEL_INTERNAL_LOG_DEBUG("[OTLP HTTP Client] DEBUG: Export " << span_count
+                                                                      << " trace span(s) success");
+        }
+        return true;
+      });
+  return opentelemetry::sdk::common::ExportResult::kSuccess;
+#else
+  opentelemetry::sdk::common::ExportResult result = http_client_->Export(service_request);
```

Comment:
Will this `Export` wait for the spans to be uploaded before returning?

Reply:
Yes, I think it's better to keep the same behaviour as before without `ENABLE_ASYNC_EXPORT` .